test/export/test_passes.py

```diff
@@ -349,7 +349,7 @@ def forward(self, x: torch.Tensor, y: torch.Tensor) -> torch.Tensor:
         y = torch.rand(3, 6)
         with self.assertRaisesRegex(
             RuntimeError,
-            "Input arg1's dimension #1 size is not equal to input arg0's dimension #1",
+            r"Input arg0.shape\[1\]'s value is not equal to input arg1.shape\[1\]"
```

Comment:
I'm not entirely sure at what point these constraints are asserted with your changes, so I'd feel more comfortable if you also tried the specified constraint in the reverse order, `dynamic_dim(y, 1) == dynamic_dim(x, 1)` to check that it works too.

Reply:
OK so I now understand what's going on better (https://github.com/pytorch/pytorch/pull/102432#discussion_r1210716420)